agent/roles/manifests/templates/install-config_yaml.j2

```diff
@@ -66,6 +69,34 @@ platform:
 {% for ingress_vip in i_vips %}
       - {{ ingress_vip }}
 {% endfor %}
+{% if agent_install_config_hosts == "true" %}
```

Comment:
Yeah this can be cleaned up. I think for now the best will be to add an additional file - install-config-bm-hosts.yaml.j2, and then choose in in the ansible tasks to use that file based on the env setting, similar to what we do for agent-config bonds https://github.com/openshift-metal3/dev-scripts/blob/master/agent/roles/manifests/tasks/install-config.yml#L21. 
This will simplify the base template a lot. We can eventually simplify it even more by splitting out `none` platform as you indicated.

Reply:
Updated to use the platform-type to choose install-config.yaml file to use.  We can eventually use more macros to split out the common sections, will save that for a future PR.